src/Internal/Cardano/Types/Transaction.purs

```diff
@@ -598,9 +598,11 @@ instance Show MoveInstantaneousReward where
   show = genericShow
 
 instance EncodeAeson MoveInstantaneousReward where
-  encodeAeson' = case _ of
-    ToOtherPot r -> encodeAeson' $ encodeTagged' "ToOtherPot" r
-    ToStakeCreds r -> encodeAeson' $ encodeTagged' "ToStakeCreds" r
+  encodeAeson = case _ of
+    ToOtherPot r -> encodeTagged' "ToOtherPot" r
+      { pot = unsafePartial partialFiniteNumber r.pot }
```

Comment:
It's a direct port. Now unsafety just become explicit.

Reply:
We actually, should not use partial here. But we can use it, because it "worked fine before". So, I don't know how to explain this, I can, probably, put something like this message in comments.

The right place to perform finiteness check are places where we construct these types. 

But, this PR is an adaptation of existing codebase to new version of "aeson", which unlike the previous one does not allow to serialize arbitrary Number. You should provide the proof of finiteness `Finite Number`. 

I am not sure this PR is a good place to "fix" something or, add new features. The goal was to preserve behavior as-is. 

So, it was already partial before, but this partiality was implicit: It just silently put "null" if number is Infinity or NaN. If we really want to get rid of partial here, we better introduce a new PR.